### PR TITLE
Dev Proxy support

### PR DESCRIPTION
**Pull request by alecuba16**
Short Description:  Proxy support implemented with exports.

Fixes:
-  Cannot log-in on banned cloud services by using a proxy.
- 
-

### CODE REVIEW

File: pokecli.py [hunk 1]
```diff
@@ -322,14 +322,38 @@ def init_config():
...
     config = parser.parse_args()
     if not config.username and 'username' not in load:
         config.username = raw_input("Username: ")
     if not config.password and 'password' not in load:
         config.password = getpass("Password: ")
-
+    if config.proxy_ip and len(config.proxy_ip)>0 and config.proxy_ip.count('.') == 3 and  all(0<=int(num)<256 for num in config.proxy_ip.rstrip().split('.')):
+        if config.proxy_port and int(config.proxy_port )<65536 and  int(config.proxy_port )>1:
+            os.environ['http_proxy']="http://"+config.proxy_ip+":"+str(config.proxy_port)+"/"
+            os.environ['https_proxy']="http://"+config.proxy_ip+":"+str(config.proxy_port)+"/"
```
Comment (douglascamata) [inline pokecli.py]: Extract this to a method, please.... something called `setup_proxy`

Comment (alecuba16) [inline pokecli.py]: Just moved to a simple method, check if you like it.

**Comment by douglascamata** [comment]
👍

**Comment by TheSavior** [comment]
:+1:

**Comment by mhdasding** [comment]
@alecuba16  why do the cli arguments have to be required?

**Comment by TheSavior** [comment]
@mhdasding, they shouldn't be. Whoops

**Comment by alecuba16** [comment]
@mhdasding True, they should be optional, fixed.

**Comment by Carlito007** [comment]
hi the proxy future is ready ?

**Pull request merged**